extension/js/common/core/expiration-cache.ts

```diff
@@ -0,0 +1,32 @@
+/* ©️ 2016 - present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com */
+
+/**
+ * Cache, keeping entries for limited duration
+ */
+export class ExpirationCache {
+  private cache: { [key: string]: { value: string, expiration: number } } = {};
+
+  constructor(public EXPIRATION_TICKS: number) {
+  }
+
+  public set = async (key: string, passphrase: string | undefined) => {
+    if (passphrase) {
+      this.cache[key] = { value: passphrase, expiration: Date.now() + this.EXPIRATION_TICKS };
+    } else {
+      delete this.cache[key];
+    }
+  }
+
+  public get = (key: string): string | undefined => {
+    const found = this.cache[key];
+    if (found) {
+      if (found.expiration > Date.now()) {
+        return found.value;
+      } else {
+        // expired, so delete it and return as if not found
+        delete this.cache[key];
```

Comment:
The simplest implementation. Entries are deleted on `get` request. We can also have a look at packages out there, like https://www.npmjs.com/package/expiring-cache

Reply:
Should additionally set up a setInterval that will go through them minute, for example. That way it's not actually kept in memory over the time limit.

Reply:
Done